src/sentry/api/bases/user.py

```diff
@@ -65,10 +67,19 @@ def convert_args(self, request: Request, user_id, *args, **kwargs):
                 raise ResourceDoesNotExist
             user_id = request.user.id
 
-        try:
-            user = User.objects.get(id=user_id)
-        except (User.DoesNotExist, ValueError):
-            raise ResourceDoesNotExist
+        # This split decreases code churn in every control silo endpoint controller by returning
```

Comment:
I'd like to discuss this in depth.  I don't disagree in theory, but I really worry about any shared code paths that must constantly remain splitting the difference between full ORM uses and Rpc Users.  No one will test both objects.

Reply:
We already see this as we carefully roll out changes -- codepaths that split the orm and the rpc object do exist, but they are fragile.  Generally things have gotten more solid as we close the loop such that code paths consistently use one type.

Reply:
I'm definitely open to feedback here. In my mind, the end state we end up in is a basically totally isolated set of endpoints/models in each silo. Probably in different, parallel folder structures in the monorepo. In this model, the control silo endpoints having full access to a user ORM is clearer, more idiomatic, and reduces the cognitive overhead for our fellow developers. Obviously it may take some time to get there.

> and then revise how avatars are saved so that we don't rely on the ORM model relations and methods?
Note that this isn't just about avatars. If we _do_ change this to use the service method and return RPCUsers, it necessitates touching almost every controller endpoint. I started down that path and pulled back because it was a deep rabbit hole.

Reply:
I definitely agree avoiding a deep rabbit hole.  I'd still like to personally open up on this some more and see what can be done.  I'll take a closer look and let you know.

Reply:
You know what... just do what you need to do.  This is always something we can change after the fact anyways.